### ISSUE TITLE
[bug] can't capture multiple match from minified scripts

### ISSUE THREAD

**Issue by DrSensor** [ast-grep/ast-grep#171]
Take this minified js file as an example
```js
import{h as M}from"./registry-6576a97f.js";import{update as O}from"./accessor-1273d0ae.js";import{r as x}from"./task-b76ba808.js";const C=(n,t,{controller:s})=>{g?.();const[o,i,e]=k.get(s),[,r]=M.get(n.prototype),[_,j]=o(r,t),d=Math.min(...j.map(p=>p.length)),c=[...i(d,t)];g=x(()=>{for(const p of _)for(let a=d,m=c.pop();a--;){if(a===m){m=c.pop();continue}const b=r[p];O(b.t,b.s,a)}}),e(c)};class l{#s;#t={};#o=1/0;#i=[];#e=new Map;stop(){const t=this.#s;return t!==void 0&&(this.#o=t),t}skip(t){return t??=this.#s,t!==void 0&&(this.#i.push(t),this.#r(t)),t}constructor(){const t=(i,e)=>(f=this.#t,u=i,e(this.#s=0,I),f=u=void 0,[Object.keys(this.#t),Object.values(this.#t)]),s=(i,e)=>{for(let r;i>(r=++this.#s)&&this.#o>r;)e(r,this.#t);return this.#s=void 0,this.#o=1/0,new Set(this.#i)},o=i=>{for(const e of i)this.#n(e);this.#t={},this.#i=[]};k.set(this,[t,s,o])}#r(t){const s={};for(const o in this.#t){const i=this.#t[o];s[o]=i[t]}this.#e.set(t,s)}#n(t){const s=this.#e.get(t);for(const o in this.#t)this.#t[o][t]=s[o]}}let g,f,u;const I=new Proxy({},{get:(n,t)=>f[t]=u[t].t}),k=new WeakMap,v=(n,t,s={controller:h})=>C(n,t,s),h=new l,w=h.skip.bind(h),y=h.stop.bind(h),z=Object.freeze({__proto__:null,Controller:l,for:v,skip:w,stop:y});export{l as I,y as a,v as b,z as i,w as s};
```
when queried with
```console
asg run -p 'import $A from "$B"' dist/
```
it only highlight `import{h as M}from"./registry-6576a97f.js";` and `import{r as x}from"./task-b76ba808.js";` but not `import{update as O}from"./accessor-1273d0ae.js";`

![image](https://user-images.githubusercontent.com/4953069/211763033-227914c9-a832-4861-9976-9681de53fc8c.png)

Tested using ast-grep 0.1.14

**Comment by HerringtonDarkholme** [comment]
Thanks for reporting!  I can reproduce it locally. But not on playground. I believe it is a bug on command line.

https://ast-grep.github.io/playground.html#eyJtb2RlIjoiUGF0Y2giLCJsYW5nIjoiamF2YXNjcmlwdCIsInF1ZXJ5IjoiaW1wb3J0ICRBIGZyb20gXCIkQlwiIiwiY29uZmlnIjoiIyBDb25maWd1cmUgUnVsZSBpbiBZQU1MXG5ydWxlOlxuICBhbGw6XG4gICAgLSBwYXR0ZXJuOiAkQVxuICAgIC0ga2luZDogdGVtcGxhdGVfc3Vic3RpdHV0aW9uXG4gICAgLSBpbnNpZGU6XG4gICAgICAgIHBhdHRlcm46IGltcG9ydCgkKVxuY29uc3RyYWludHM6XG4gICMgTUVUQV9WQVI6IHBhdHRlcm4iLCJzb3VyY2UiOiJpbXBvcnR7aCBhcyBNfWZyb21cIi4vcmVnaXN0cnktNjU3NmE5N2YuanNcIjtpbXBvcnR7dXBkYXRlIGFzIE99ZnJvbVwiLi9hY2Nlc3Nvci0xMjczZDBhZS5qc1wiO2ltcG9ydHtyIGFzIHh9ZnJvbVwiLi90YXNrLWI3NmJhODA4LmpzXCI7Y29uc3QgQz0obix0LHtjb250cm9sbGVyOnN9KT0+e2c/LigpO2NvbnN0W28saSxlXT1rLmdldChzKSxbLHJdPU0uZ2V0KG4ucHJvdG90eXBlKSxbXyxqXT1vKHIsdCksZD1NYXRoLm1pbiguLi5qLm1hcChwPT5wLmxlbmd0aCkpLGM9Wy4uLmkoZCx0KV07Zz14KCgpPT57Zm9yKGNvbnN0IHAgb2YgXylmb3IobGV0IGE9ZCxtPWMucG9wKCk7YS0tOyl7aWYoYT09PW0pe209Yy5wb3AoKTtjb250aW51ZX1jb25zdCBiPXJbcF07TyhiLnQsYi5zLGEpfX0pLGUoYyl9O2NsYXNzIGx7I3M7I3Q9e307I289MS8wOyNpPVtdOyNlPW5ldyBNYXA7c3RvcCgpe2NvbnN0IHQ9dGhpcy4jcztyZXR1cm4gdCE9PXZvaWQgMCYmKHRoaXMuI289dCksdH1za2lwKHQpe3JldHVybiB0Pz89dGhpcy4jcyx0IT09dm9pZCAwJiYodGhpcy4jaS5wdXNoKHQpLHRoaXMuI3IodCkpLHR9Y29uc3RydWN0b3IoKXtjb25zdCB0PShpLGUpPT4oZj10aGlzLiN0LHU9aSxlKHRoaXMuI3M9MCxJKSxmPXU9dm9pZCAwLFtPYmplY3Qua2V5cyh0aGlzLiN0KSxPYmplY3QudmFsdWVzKHRoaXMuI3QpXSkscz0oaSxlKT0+e2ZvcihsZXQgcjtpPihyPSsrdGhpcy4jcykmJnRoaXMuI28+cjspZShyLHRoaXMuI3QpO3JldHVybiB0aGlzLiNzPXZvaWQgMCx0aGlzLiNvPTEvMCxuZXcgU2V0KHRoaXMuI2kpfSxvPWk9Pntmb3IoY29uc3QgZSBvZiBpKXRoaXMuI24oZSk7dGhpcy4jdD17fSx0aGlzLiNpPVtdfTtrLnNldCh0aGlzLFt0LHMsb10pfSNyKHQpe2NvbnN0IHM9e307Zm9yKGNvbnN0IG8gaW4gdGhpcy4jdCl7Y29uc3QgaT10aGlzLiN0W29dO3Nbb109aVt0XX10aGlzLiNlLnNldCh0LHMpfSNuKHQpe2NvbnN0IHM9dGhpcy4jZS5nZXQodCk7Zm9yKGNvbnN0IG8gaW4gdGhpcy4jdCl0aGlzLiN0W29dW3RdPXNbb119fWxldCBnLGYsdTtjb25zdCBJPW5ldyBQcm94eSh7fSx7Z2V0OihuLHQpPT5mW3RdPXVbdF0udH0pLGs9bmV3IFdlYWtNYXAsdj0obix0LHM9e2NvbnRyb2xsZXI6aH0pPT5DKG4sdCxzKSxoPW5ldyBsLHc9aC5za2lwLmJpbmQoaCkseT1oLnN0b3AuYmluZChoKSx6PU9iamVjdC5mcmVlemUoe19fcHJvdG9fXzpudWxsLENvbnRyb2xsZXI6bCxmb3I6dixza2lwOncsc3RvcDp5fSk7ZXhwb3J0e2wgYXMgSSx5IGFzIGEsdiBhcyBiLHogYXMgaSx3IGFzIHN9OyJ9


BTW, `asg` sounds a good name. Are you using an alias to avoid `sg` command conflict?

**Comment by HerringtonDarkholme** [comment]
I tested with json flag
```
asg run -p 'import $A from "$B"' dist/ --json
```

The output does have three matches.

<img width="701" alt="image" src="https://user-images.githubusercontent.com/2883231/211811906-4dd8cf65-f678-4a1b-8062-586b50afcd9a.png">

But colored output is not correct

<img width="973" alt="image" src="https://user-images.githubusercontent.com/2883231/211812020-da1c257a-d154-4e3c-bd16-f16f07d8fe9d.png">

**Comment by DrSensor** [comment]
Didn't knew there is `--json` flag. I think I can make workaround with that by pipe it to `jq` and `fzf`.

---
> BTW, `asg` sounds a good name. Are you using an alias to avoid `sg` command conflict?

Yes

**Comment by HerringtonDarkholme** [comment]
fixed in 0.1.16
